src/settings/toml/tests/deploy_target.rs

```diff
@@ -335,3 +335,540 @@ fn it_errors_on_deploy_target_routes_and_workers_dev_true() {
 
     assert!(manifest.deploy_target(environment).is_err());
 }
+
+// ENVIRONMENT TESTS
+// Top level empty
+#[test]
+fn when_top_level_empty_env_empty() {
+    let script = "top_level_empty_env_empty";
+    let env_name = "test";
+    let env_config = EnvConfig::default();
+
+    let test_toml = WranglerToml::webpack_with_env(script, env_name, env_config);
+    let toml_string = toml::to_string(&test_toml).unwrap();
+    let manifest = Manifest::new_from_string(toml_string).unwrap();
+
+    let actual_deploy_target = manifest.deploy_target(Some(env_name));
+
+    assert!(actual_deploy_target.is_err());
+
+    // if env only includes zone id, error
+    let mut env_config = EnvConfig::default();
+    env_config.zone_id = Some("samplezoneid");
+
+    let test_toml = WranglerToml::webpack_with_env(script, env_name, env_config);
+    let toml_string = toml::to_string(&test_toml).unwrap();
+    let manifest = Manifest::new_from_string(toml_string).unwrap();
+
+    let actual_deploy_target = manifest.deploy_target(Some(env_name));
+
+    assert!(actual_deploy_target.is_err());
+}
+
+#[test]
+fn when_top_level_empty_zoneless_env() {
+    let env_name = "test";
+
+    // TODO: THIS TEST SHOULD PASS
+    // when env.workers_dev = false
+    let env_config = EnvConfig::zoneless(false);
```

Comment:
Shouldn't this then require a zone and a route(s) to pass, then?

Reply:
the assert is that this errors...

Reply:
Yup! Just responding to the TODO above. I still think this test should fail, as it does now.

Reply:
when you say "this test should fail", you mean this condition should error? the test passing == the condition errors. it's confusing but i just want to  make sure i follow